apiserver/facades/agent/machine/machiner.go

```diff
@@ -144,31 +160,23 @@ func (api *MachinerAPI) SetMachineAddresses(ctx context.Context, args params.Set
 	return results, nil
 }
 
-// Jobs returns the jobs assigned to the given entities.
-func (api *MachinerAPI) Jobs(ctx context.Context, args params.Entities) (params.JobsResults, error) {
-	result := params.JobsResults{
-		Results: make([]params.JobsResult, len(args.Entities)),
-	}
+// SetMachineAddresses is not supported in MachinerAPI at version 5.
+func (api *MachinerAPIv5) SetMachineAddresses(ctx context.Context, args params.SetMachinesAddresses) (params.ErrorResults, error) {
+	return params.ErrorResults{}, errors.NotSupported
+}
 
-	canRead, err := api.getCanRead()
-	if err != nil {
-		return result, err
-	}
+// Jobs is not supported in MachinerAPI at version 5.
```

Comment:
I would prefer not to make this through an error.

For version 5, keep `params.JobsResults` and mark it deprecated. Call `IsMachineController` and return "api-server", "host-units" if true, otherwise just "host-units".

No need to keep the other types, just pack the strings directly.